packages/big-design/src/components/Modal/Modal.tsx

```diff
@@ -136,18 +116,28 @@ export class Modal extends React.PureComponent<ModalProps, ModalState> {
     );
   }
 
-  private renderChildren() {
-    const { children } = this.props;
+  private renderHeader() {
+    const { header, withHeaderBorder = false } = this.props;
 
-    return React.Children.map(children, child => {
-      if (React.isValidElement(child) && child.type === Modal.Header) {
-        return React.cloneElement(child as React.ReactElement<HeadingProps>, {
-          id: this.headerUniqueId,
-        });
-      }
+    return (
+      header && (
+        <StyledModalHeader id={this.headerUniqueId} withBorder={withHeaderBorder}>
+          <H2 margin="none">{header}</H2>
+        </StyledModalHeader>
+      )
+    );
+  }
+
+  private renderActions() {
+    const { actions, withActionsBorder = false } = this.props;
 
-      return child;
-    });
+    return (
+      actions && (
+        <StyledModalActions justifyContent="flex-end" withBorder={withActionsBorder}>
+          {actions}
+        </StyledModalActions>
+      )
+    );
   }
```

Comment:
To go along with @deini runtime check for strings, should we be consistent and do the same for `Buttons` within this `renderActions` method?

Reply:
Makes sense however how can you test for `Buttons` when you receive a set of actions?

For example in:
```
  const actions = (
    <>
      <Button variant="subtle">Cancel</Button>
      <Button>Apply</Button>
    </>
  );
```
In this case `React.Children.map` wouldn't iterate over the buttons.